src/shapes/object.class.js

```diff
@@ -448,6 +448,35 @@
      */
     clipTo:                   null,
 
+
+    /**
+     * When not null, will not allow the width of the object to be smaller than the value
+     * @type Number
+     * @default
+     */
+    minWidth:                null,
+
+    /**
+     * When not null, will not allow the width of the object to be larger than the value
+     * @type Number
+     * @default
+     */
+    maxWidth:                null,
+
+    /**
+     * When not null, will not allow the height of the object to be smaller than the value
+     * @type Number
+     * @default
+     */
+    minHeight:                null,
+
+    /**
+     * When not null, will not allow the height of the object to be larger than the value
+     * @type Number
+     * @default
+     */
+    maxHeight:                null,
```

Comment:
if we ask for numbers as type, we do not want to initialize as null. you can initialize as Number.INFINITY as you did in the functions. and let the getters out.